torch/csrc/jit/script/python_sugared_value.cpp

```diff
@@ -544,6 +544,11 @@ std::shared_ptr<SugaredValue> toSugaredValue(
   } else if (
       obj.ptr() == py::module::import("torch.jit").attr("annotate").ptr()) {
     return SpecialFormValue::create(prim::annotate);
+#ifdef USE_DISTRIBUTED
+  } else if (
+      obj.ptr() == py::module::import("torch.distributed.rpc.api").attr("_invoke_rpc_torchscript").ptr()) {
```

Comment:
could we create prim::async node based on rpc_async api here? considering rpc_async signature is almost the same as _invoke_rpc_torchscript now

Reply:
I guess we can,

but we need to move these logic from Python to C++.
- autograd profiling
- to_worker_info(to)
- None arg or Non kwargs handling.

This will cause duplicate code paths, one implemented in Python and the other implemented in C++.

Reply:
@zhaojuanmao 

I changed the dispatching point to `torch.distributed.rpc.api.rpc_async`.

In order to support allowing users to not passing args and kwargs. 
I added more logic to IR emitting and JIT operator code, and added tests cases for the 2 cases.
- kwargs not passed.
- args + kwargs not passed.